Migrate Workflows section to 6 0 (#11701)
storybookjs/storybook: docs/workflows/stories-for-multiple-components.md

---
title: 'Stories for multiple components'
---

> As discussed in an [earlier section](../writing-stories/introduction#stories-for-two-or-more-components), if you have sets of components that are designed to work together, it is convenient to write stories that render two or more components at once.

Reviewer: ```suggestion
It's useful to write stories that [render two or more components](../writing-stories/introduction#stories-for-two-or-more-components) at once if those components are designed to work together. For example, `ButtonGroups`, `Lists`, and `Page` components. 
```

Updated to not assume that the reader read the previous section.

Author: Marking this as resolved, applied the change

```js
// List.story.js

const Template = (args) => <List {...args} />

export const OneItem = Template.bind({});
OneItem.args = {
  children: <Unchecked {...Unchecked.args} />
}
```

Now that `children` is an arg, we can potentially reuse it in another story. As things stand (we hope to improve this soon) you cannot edit children in a control yet.

### Creating a Template Component

Another option that is more “data”-based is to create a special “story-generating” template component:

```js
// List.story.js

import React from 'react';
import List from './List';
import ListItem from './ListItem';
import { Unchecked } from './ListItem.stories';

const ListTemplate = ({ items, ...args }) => (
  <List>
    {items.map(item => <ListItem {...item} />)}
  </List>
);

export const Empty = ListTemplate.bind({});
Empty.args = { items: [] };

export const OneItem = ListTemplate.bind({});
OneItem.args = { items: [Unchecked.args] };
```

This approach is a little more complex to setup, but it means you can more easily reuse the `args` to each story in a composite component. It also means that you can alter the args to the component with the Controls addon:

> <div style="background-color:#F8FAFC">

Reviewer: @tmeasday what did you have in mind here? I don't have the code on my machine to screenshot this.
